edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionManager.cs

```diff
@@ -265,12 +266,8 @@ await device.CloudConnection.Filter(cp => cp.IsActive)
                 case CloudConnectionStatus.Disconnected:
                     Events.InvokingCloudConnectionLostEvent(device.Identity);
                     this.CloudConnectionLost?.Invoke(this, device.Identity);
-                    await device.CloudConnection.Filter(cp => cp.IsActive).ForEachAsync(
-                        cp =>
-                        {
-                            Events.CloudConnectionLostClosingClient(device.Identity);
-                            return cp.CloseAsync();
-                        });
+                    // Closing all connections so new connections are created a network disconnection
+                    await this.CloseAllCloudConnections();
```

Comment:
Is this correct? This is the callback for each individual connection, right?

Reply:
Yes, this is the callback for each connection. I added IDeviceConnectivityManager to Connectionmanager and subscribe to DeviceDisconnected